lib/mapUtils.js

```diff
@@ -43,3 +43,156 @@ export async function getProductData(store) {
 export const getMaxWidth = function sync(screenWidth, ebt, seeProducts) {
   return screenWidth - 32 - 12 + (ebt ? -48 : 0) + (seeProducts ? -130 : 0);
 };
+
+export function constructHoursDict(hours) {
+  const daysOfTheWeek = {
+    Sun: 'Sunday',
+    Mon: 'Monday',
+    Tu: 'Tuesday',
+    Wed: 'Wednesday',
+    Thurs: 'Thursday',
+    Fri: 'Friday',
+    Sat: 'Saturday',
+  };
+  const daysOfTheWeekAbbrev = Object.keys(daysOfTheWeek);
+  const daysOfTheWeekFull = Object.values(daysOfTheWeek);
+  const hoursStrSplit = hours.split(' ');
+
+  // Case: Hours are daily if there are only two strings
+  // e.g. 8am-7pm Daily, 7am-Midnight Daily
+
+  // We expect hoursPerDayDict to look like {'Sunday': '9am-9pm', 'Monday: '7am-12am', ...}
```

Comment:
OH also since we changed the input format Midnight to 12am, could we switch that back to display 'Midnight' if it is 12am? Could probably be done with a `.replace` or something....just bc I 12am/midnight/noon always confuses me and I understand why they originally wanted it to say Midnight 😅

Reply:
Wait so if they input `7am-12am` we want the app to show `7am-Midnight` instead?